custom_components/isy994/services.py

```diff
@@ -0,0 +1,232 @@
+            if address and isy.nodes.get_by_id(address) is not None:
+                _LOGGER.debug(
+                    "Requesting query of device %s on ISY %s",
+                    address,
+                    isy.configuration["uuid"],
+                )
+                # TODO: Enable once PyISY is updated with PR#94
+                # await hass.async_add_executor_job(ft.partial(isy.query, address))
+                return
+            _LOGGER.debug(
+                "Requesting system query of ISY %s", isy.configuration["uuid"]
+            )
+            # TODO: Enable once PyISY is updated with PR#94
+            # await hass.async_add_executor_job(isy.query)
+
+    hass.services.async_register(
+        domain=DOMAIN,
+        service=SERVICE_SYSTEM_QUERY,
+        service_func=async_system_query_service_handler,
+        schema=SERVICE_SYSTEM_QUERY_SCHEMA,
+    )
+
+
+@callback
+def async_unload_services(hass: HomeAssistantType):
+    """Unload services for the ISY integration."""
+    if hass.data[DOMAIN]:
+        # There is still another config entry for this domain, don't remove services.
+        return
+
+    _LOGGER.info("Unloading ISY994 Services.")
+    hass.services.async_remove(domain=DOMAIN, service=SERVICE_SYSTEM_QUERY)
+
+
+@callback
+def async_setup_device_services(hass: HomeAssistantType):
+    """Create device-specific services for the ISY Integration."""
+    platform = entity_platform.current_platform.get()
+
+    async def device_service_handler(entity, service_call):
+        """Handle device-specific service calls."""
+        # pylint: disable=protected-access
+        if not hasattr(entity, "_node") or not hasattr(
+            entity._node, service_call.service
+        ):
+            _LOGGER.warning(
+                "Invalid Service Call %s for device %s.",
+                service_call.service,
+                entity.entity_id,
+            )
+            return
+
+        if service_call.service == SERVICE_SEND_COMMAND:
+            command = service_call.data.get(ATTR_COMMAND)
+            value = service_call.data.get(ATTR_VALUE)
+            parameters = service_call.data.get(ATTR_PARAMETERS)
+            uom = service_call.data.get(ATTR_UOM)
+            _LOGGER.debug("Sending command %s to device %s.", command, entity.entity_id)
+            await hass.async_add_executor_job(
+                ft.partial(entity._node.send_cmd, command, value, uom, parameters)
+            )
+            return
+
+        _LOGGER.debug(
+            "Sending command %s to device %s.", service_call.service, entity.entity_id
+        )
+        await hass.async_add_executor_job(getattr(entity._node, service_call.service))
+
+    platform.async_register_entity_service(
+        SERVICE_SEND_COMMAND, SERVICE_SEND_COMMAND_SCHEMA, device_service_handler
+    )
+    platform.async_register_entity_service(
+        SERVICE_BEEP, SERVICE_ENTITY_SCHEMA, device_service_handler
```

Comment:
Probably doesn't make much difference here: if you pass an dict it gets auto converted to schema.
```
    @callback
    def async_register_entity_service(self, name, schema, func, required_features=None):
        """Register an entity service.

        Services will automatically be shared by all platforms of the same domain.
        """
.....
        if isinstance(schema, dict):
            schema = cv.make_entity_service_schema(schema)
```